public/app/features/logs/components/panel/usePopoverMenu.ts

```diff
@@ -0,0 +1,115 @@
+import { useCallback, useRef, useState, MouseEvent } from 'react';
+
+import { config } from '@grafana/runtime';
+
+import { disablePopoverMenu, enablePopoverMenu, isPopoverMenuDisabled, targetIsElement } from '../../utils';
+import { PopoverStateType } from '../LogRows';
+
+import { useLogListContext } from './LogListContext';
+import { LogListModel } from './processing';
+
```

Comment:
This is a nice abstraction. I usually put stuff like this in a `/hooks` folder or something. Does any concept exist like that in core?